assets/src/components/mapPage.tsx

```diff
@@ -0,0 +1,116 @@
+import { Socket } from "phoenix"
+import React, { ReactElement, useContext, useState } from "react"
+import { SocketContext } from "../contexts/socketContext"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import useSearchResults from "../hooks/useSearchResults"
+import { useTripShape } from "../hooks/useShapes"
+import { isVehicle } from "../models/vehicle"
+import { Vehicle, VehicleOrGhost } from "../realtime"
+import { SearchPageState } from "../state/searchPageState"
+import Map from "./map"
+import RecentSearches from "./recentSearches"
+import SearchForm from "./searchForm"
+import SearchResults from "./searchResults"
+
+enum MobileDisplay {
+  List = 1,
+  Map,
+}
+
+const thereIsAnActiveSearch = (
+  vehicles: VehicleOrGhost[] | null,
+  searchPageState: SearchPageState
+): boolean => vehicles !== null && searchPageState.isActive
+
+const filterVehicles = (
+  vehiclesOrGhosts: VehicleOrGhost[] | null
+): Vehicle[] => {
+  return vehiclesOrGhosts === null
+    ? []
+    : (vehiclesOrGhosts.filter((vog) => isVehicle(vog)) as Vehicle[])
+}
+
+const ToggleMobileDisplayButton = ({
+  mobileDisplay,
+  onToggleMobileDisplay,
+}: {
+  mobileDisplay: MobileDisplay
+  onToggleMobileDisplay: () => void
+}) => {
+  const otherDisplayName = mobileDisplay === MobileDisplay.List ? "map" : "list"
+
+  return (
+    <button
+      className="m-search-page__toggle-mobile-display-button"
+      onClick={onToggleMobileDisplay}
+    >
+      Show {otherDisplayName} instead
+    </button>
+  )
+}
+
+const MapPage = (): ReactElement<HTMLDivElement> => {
+  const [{ searchPageState, mobileMenuIsOpen }] =
+    useContext(StateDispatchContext)
+
+  const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
+  const vehicles: VehicleOrGhost[] | null = useSearchResults(
+    socket,
+    searchPageState.isActive ? searchPageState.query : null
+  )
+  const onlyVehicles: Vehicle[] = filterVehicles(vehicles)
+  const [mobileDisplay, setMobileDisplay] = useState(MobileDisplay.List)
+  const [selectedVehicle, setSelectedVehicle] = useState<Vehicle | null>(null)
+  const selectedVehicleShapes = useTripShape(selectedVehicle?.tripId || null)
```

Comment:
This is the key logic change vs `searchPage`: maintaining new state for `selectedVehicle` & fetching the trip shape rather than displaying the VPP.